inntektservice/src/main/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/inntektservice/InntektService.kt

```diff
@@ -175,7 +181,47 @@ class InntektService(
         }
     }
 
-    override fun terminate(fail: Fail) {
+    override fun onError(message: JsonMessage, fail: Fail) {
+        val utloesendeBehov = Key.BEHOV.lesOrNull(BehovType.serializer(), fail.utloesendeMelding.toMap())
+
+        if (utloesendeBehov == BehovType.INNTEKT) {
+            val feilmelding = Feilmelding(
+                "Vi har problemer med å hente inntektsopplysninger. Du kan legge inn beregnet månedsinntekt manuelt, eller prøv igjen senere.",
+                datafelt = Key.INNTEKT
+            )
+
+            "Legger til feilmelding: '${feilmelding.melding}'".also {
+                logger.error(it)
+                sikkerLogger.error(it)
+            }
+
+            RedisKey.of(fail.transaksjonId, feilmelding).write(
+                FeilReport(
+                    mutableListOf(feilmelding)
+                ).toJson(FeilReport.serializer())
+            )
+
+            RedisKey.of(fail.transaksjonId, Key.INNTEKT).write(JsonObject(emptyMap()))
+
+            // TODO bruk finalize (sjekk andre servicer for tilsvarende feil)
```

Comment:
Skjønner ikke denne helt - hvorfor kalles inProgress her, og hva med TODO'en

Reply:
Legger til TODO-en fordi jeg ikke ønsket å endre for mye av oppførselen til servicene med denne PR-en, men heller fokusere på det som spinner under panseret. Så `inProgress` kalles fordi det var original oppførsel. Se https://github.com/navikt/helsearbeidsgiver-inntektsmelding/pull/388/files#diff-babcc7677280f328041f0947da9bc0f396dd0deb964b07106d1d30d7c49ebf1bL219-L249.

Reply:
Ah, skjønte det nå - litt vanskelig å lese diffen